gsi/gssapi/source/test/gssapi_anonymous_test.c

```diff
@@ -60,8 +60,15 @@ int main()
     
     /* acquire the credential */
 
+#if OPENSSL_VERSION_NUMBER < 0x30000000L
```

Comment:
@msalle:
OpenSSL uses [`#  ifndef OPENSSL_NO_DEPRECATED_3_0`](https://github.com/openssl/openssl/blob/b8be229dab036b26de8830444bf2beb82e71f50e/include/openssl/crypto.h.in#L361), not sure what's the better way. But I definitely find the version check used in your PR much better comprehensible.

Reply:
The [corresponding documentation](https://www.openssl.org/docs/manmaster/man7/OPENSSL_NO_DEPRECATED.html) also doesn't make it clearer IMHO.

Reply:
I think for us it's much better to stick to a direct check against the version. That has always worked and is therefore more reliable.
For background, see e.g. https://github.com/openssl/openssl/blob/master/include/openssl/crypto.h.in#L361-L363 and https://github.com/openssl/openssl/blob/master/include/openssl/macros.h#L159-L190 plus https://github.com/openssl/openssl/blob/master/doc/internal/man7/deprecation.pod

Reply:
I fully agree and it's easier to understand anyhow.